backend/de.metas.swat/de.metas.swat.base/src/main/java/org/adempiere/inout/util/ShipmentScheduleQtyOnHandStorageHolder.java

```diff
@@ -0,0 +1,40 @@
+package org.adempiere.inout.util;
+
+import com.google.common.collect.ImmutableList;
+import de.metas.inoutcandidate.api.OlAndSched;
+import de.metas.inoutcandidate.model.I_M_ShipmentSchedule;
+import de.metas.material.cockpit.stock.StockDataQuery;
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Loads stock details which are relevant to given {@link I_M_ShipmentSchedule}s.
+ * Allows to change (in memory!) the qtyOnHand.
+ */
+@AllArgsConstructor(staticName="of")
+public class ShipmentScheduleQtyOnHandStorageHolder
+{
+	private final ImmutableList<IShipmentScheduleQtyOnHandStorage> storages;
+
+	@NonNull
+	public Optional<StockDataQuery> toQuery(@NonNull final I_M_ShipmentSchedule sched)
+	{
+		return storages.stream()
+				.map(storage -> storage.toQuery(sched))
+				.filter(Objects::nonNull)
+				.findFirst();
+	}
+
+	public ShipmentScheduleAvailableStock getStockDetailsMatching(@NonNull final OlAndSched olAndSched)
+	{
+		return ShipmentScheduleAvailableStock.of(storages.stream()
+				.map(storage -> storage.getStockDetailsMatching(olAndSched.getSched()))
+				.flatMap(List::stream)
+				.collect(ImmutableList.toImmutableList()));
+	}
```

Comment:
Just an idea, maybe it would be worth validating that we are not talking about the same HU from two different sources (reserved vs available)... in theory we shouldn't, but just in case, I think it would be nice to fail here with an explicit error than somewhere else down the road

Reply:
@adi-stefan  let's talk about this tomorrow... maybe we can come up with a simple solution ... something like a generated string key....

Reply:
as discussed, we can't. Added implSpec to `org.adempiere.inout.util.IShipmentScheduleQtyOnHandStorage#getStockDetailsMatching` as to document this shortcoming.